### PR TITLE
fix: mount kubelet config to `/var/lib/kubelet` for non-rbac deployment

### PR DESCRIPTION
**Pull request by bcho**
**Reason for Change**:

AKS is rolling out the TLS bootstrapping change. Before this change, AKS cluster agent node uses kubeconfig as follow:

```yaml
apiVersion: v1
kind: Config
clusters:
- name: localcluster
  cluster:
    certificate-authority: /etc/kubernetes/certs/ca.crt
    server: https://<api-server-address>:443
users:
- name: client
  user:
    client-certificate: /etc/kubernetes/certs/client.crt
    client-key: /etc/kubernetes/certs/client.key
contexts:
- context:
    cluster: localcluster
    user: client
  name: localclustercontext
current-context: localclustercontext
```

After enabling TLS bootstrapping, the agent node will get the kubeconfig as follow:

```yaml
apiVersion: v1
clusters:
- cluster:
    certificate-authority: /etc/kubernetes/certs/ca.crt
    server: https://<api-server-address>:443
  name: default-cluster
contexts:
- context:
    cluster: default-cluster
    namespace: default
    user: default-auth
  name: default-context
current-context: default-context
kind: Config
preferences: {}
users:
- name: default-auth
  user:
    client-certificate: /var/lib/kubelet/pki/kubelet-client-current.pem
    client-key: /var/lib/kubelet/pki/kubelet-client-current.pem
```

As we can see, in the new mode, kubelet's kubeconfig references the client certificates in `/var/lib/kubelet/pki` instead of `/etc/kubernetes/certs`. 

Since we are mounting the kubeconfig from host for those RBAC-disabled cluster, I think it's better to maintain the mount path structure for the kubeconfig (i.e. mounting `/var/lib/kubelet` from host to `/var/lib/kubelet` inside the pod) - with this, we can avoiding breaking MIC for certificates change.

I had verified this change in two AKS RBAC-disabled clusters, using with/out TLS bootstrapping setup.

**Requirements**

- [x] squashed commits
- [x] included documentation
- [ ] added unit tests and e2e tests (if applicable). See [test standard](https://github.com/Azure/aad-pod-identity/blob/master/CONTRIBUTING.md#test-standard) for more details.
- [x] ran `make precommit`

**Issue Fixed**:

**Please answer the following questions with yes/no**:

Does this change contain code from or inspired by another project? If so, did you notify the maintainers and provide attribution?

- [ ] yes
- [x] no

**Notes for Reviewers**:

### CODE REVIEW

**Review by aramase — COMMENTED** [review]
@bcho All the changes are to be made in [manifest_staging/deploy](https://github.com/Azure/aad-pod-identity/tree/master/manifest_staging/deploy) folder and will be promoted to `deploy` folder at the time of release.

File: deploy/infra/deployment.yaml [hunk 1]
```diff
@@ -490,8 +490,8 @@ spec:
             cpu: 100m
             memory: 256Mi
         volumeMounts:
-          - name: kubeconfig
-            mountPath: /etc/kubernetes/kubeconfig
+          - name: kubelet-config
+            mountPath: /var/lib/kubelet
```
Comment (aramase) [inline deploy/infra/deployment.yaml:494]: Why is this change required? The volume mount is only for within the pod. The actual volume that's still mounted is `/var/lib/kubelet` hostPath.

Comment (bcho) [reply deploy/infra/deployment.yaml]: I put the background in the PR description. The main reason is with TLS bootstrapping enabled, this kubeconfig will reference the client certificates in `/var/lib/kubelet/pki`. In this case, we should mount the whole `/var/lib/kubelet` into the pod.

**Comment by bcho** [comment]
Hi @aramase I updated the PR to modify the staging folder only, do we need to run `promote-staging-manifest` to promote the changes?

**Comment by aramase** [comment]
> Hi @aramase I updated the PR to modify the staging folder only, do we need to run promote-staging-manifest to promote the changes?

1. Could you also update https://github.com/Azure/aad-pod-identity/blob/master/manifest_staging/deploy/infra/noazurejson/deployment.yaml?
2. `promote-staging-manifest` will be run at the time of the release. So it's not required now. We'll include this yaml update in the `v1.8.1` release.

**Comment by bcho** [comment]
@aramase updated, PTAL

**Review by aramase — APPROVED** [review]
> I had verified this change in two AKS RBAC-disabled clusters, using with/out TLS bootstrapping setup.

@bcho Thanks for confirming the change is backward compatible.

lgtm

**Review by chewong — APPROVED** [review]
